internal/hcsoci/resources_wcow.go

```diff
@@ -151,35 +152,52 @@ func setupMounts(ctx context.Context, coi *createOptionsInternal, r *resources.R
 				}
 			}
 			l := log.G(ctx).WithField("mount", fmt.Sprintf("%+v", mount))
-			if mount.Type == "physical-disk" {
-				l.Debug("hcsshim::allocateWindowsResources Hot-adding SCSI physical disk for OCI mount")
-				scsiMount, err := coi.HostingSystem.AddSCSIPhysicalDisk(ctx, mount.Source, uvmPath, readOnly, mount.Options)
-				if err != nil {
-					return errors.Wrapf(err, "adding SCSI physical disk mount %+v", mount)
-				}
-				r.Add(scsiMount)
-			} else if mount.Type == "virtual-disk" {
-				l.Debug("hcsshim::allocateWindowsResources Hot-adding SCSI virtual disk for OCI mount")
-				scsiMount, err := coi.HostingSystem.AddSCSI(
-					ctx,
-					mount.Source,
-					uvmPath,
-					readOnly,
-					false,
-					mount.Options,
-					uvm.VMAccessTypeIndividual,
+			if mount.Type == "physical-disk" || mount.Type == "virtual-disk" || mount.Type == "extensible-virtual-disk" {
+				var (
+					scsiMount *uvm.SCSIMount
+					err       error
 				)
-				if err != nil {
-					return errors.Wrapf(err, "adding SCSI virtual disk mount %+v", mount)
+				switch mount.Type {
```

Comment:
Could we just separate these out into their own `if` statement options instead of using the if statement to catch all of them and then split them with a switch?

Reply:
We can, yes, it just means this code block is duplicated between each if clause:

```golang
				if err != nil {
					return fmt.Errorf("adding SCSI mount %+v: %w", mount, err)
				}
				r.Add(scsiMount)
				// Compute guest mounts now, and store them, so they can be added to the container doc later.
				// We do this now because we have access to the guest path through the returned mount object.
				coi.windowsAdditionalMounts = append(coi.windowsAdditionalMounts, hcsschema.MappedDirectory{
					HostPath:      scsiMount.UVMPath,
					ContainerPath: mount.Destination,
					ReadOnly:      readOnly,
				})
```

Do you feel there are readability issues with the current approach?

Reply:
I can see a stance that it makes sense to have one If Block deal with all the SCSI mounts, but I don't see a need to do that and I do think it makes the code a bit uglier. I don't feel like the code dup is an unreasonable amount, but I also am not super tied to either option, so I'm fine to accept as is if you feel it's the better option.

Reply:
I'd like to keep it this way for now, but okay with it changing in the future if we change this code around again. :)